api_tests/tests/ether_halight.ts

```diff
@@ -3,36 +3,70 @@
+
+            await alice.init();
 
-it(
-    "han-ethereum-ether-halight-lightning-bitcoin-alice-redeems-bob-redeems",
-    twoActorTest(async ({ alice, bob }) => {
-        const bodies = (await SwapFactory.newSwap(alice, bob))
-            .hanEthereumEtherHalightLightningBitcoin;
+            await alice.fund();
 
-        // Bob needs to know about the swap first because he is not buffering incoming announcements about swaps he doesn't know about
-        await bob.createSwap(bodies.bob);
-        await sleep(500);
+            // we must not wait for bob's funding because `sendpayment` on a hold-invoice is a blocking call.
+            // tslint:disable-next-line:no-floating-promises
+            bob.fund();
 
-        await alice.createSwap(bodies.alice);
+            await alice.redeem();
+            await bob.redeem();
 
-        await alice.init();
+            await sleep(2000);
 
-        await alice.fund();
+            await alice.assertBalances();
+            await bob.assertBalances();
+        })
+    );
 
-        // we must not wait for bob's funding because `sendpayment` on a hold-invoice is a blocking call.
-        // tslint:disable-next-line:no-floating-promises
-        bob.fund();
+    it(
+        "han-ethereum-ether-halight-lightning-bitcoin-alice-announces-with-wrong-peer-id",
+        twoActorTest(async ({ alice, bob }) => {
+            const bodies = (await SwapFactory.newSwap(alice, bob))
+                .hanEthereumEtherHalightLightningBitcoin;
 
-        await alice.redeem();
-        await bob.redeem();
+            // Simulate that Bob is awaiting a swap from a different peer-id than Alice node's peer-id.
+            bodies.bob.peer.peer_id =
+                "QmYyQSo1c1Ym7orWxLYvCrM2EmxFTANf8wXmmE7DWjhx5N";
+            await bob.createSwap(bodies.bob);
+            await sleep(500);
 
-        await sleep(2000);
+            await alice.createSwap(bodies.alice);
 
-        await alice.assertBalances();
-        await bob.assertBalances();
-    })
-);
+            // Due to the asynchronous nature of the announce protocol (and the fact that there is not error handling (yet)),
+            // this test only tests that the "init" action does not become available.
+            //
+            // Scenario details:
+            // 0) Alice and Bob agree on the swap params (negotiation); Alice gives Bob a peer-id that is not her node's peer-id.
+            // 1) Bob posts swap, retrieves local swap-ID
+            // 2) Alice posts swap, retrieves local swap-ID
+            //      2.1) Internally the announce protocol runs into an Error because Alice' peer-id does not match the one she gave Bob.
+            //              This Error, is however not visible to the outside.
+            //              The application is responsible for removing swaps that do not move forward after a given time.
+            const aliceResponsePromise = alice.init_does_not_become_available();
```

Comment:
I think just calling the regular `init` function and _not_ awaiting here would be sufficient.

Reply:
I don't agree with this, but maybe I have a knowledge lack here.

To my understanding the announce protocol exchanges several messages async between Alice and Bob, so after "announcing" we have to wait to be sure that init "did not become available". 
In the Rust code we only await that the announce was triggered correctly before returning the local-swap-id, but we don't await all the async messages underneath. 
I would keep this async.

Reply:
Differently asked: What is the value that `init_does_not_become_available` adds over the regular `init`?